### ISSUE TITLE
Define device settings programmatically

### ISSUE THREAD

**Issue by jghaanstra** [athombv/apps-sdk-issues#252]
Pretty much all device aspects can be added using `Homey.createDevice()` but the one thing I'm missing is the ability to programmatically define the device settings. You can add device settings using `Homey.createDevice()` but allowing the user to edit them under the device settings screen requires a hardcoded `driver.settings.compose.json` file.

I would much appreciate the ability to be able to set this programmatically during pairing as well. I have an app with 75+ drivers which is continuously growing and I'm looking for more flexibility to manage this.

**Comment by frodeheg** [comment]
Yes, indeed, I would very much like this added as well for the same reason. I did not have a use for it for small apps, but now that I am diving into an app that needs a whole well of drivers this was one of the first things I missed.

**Comment by jghaanstra** [comment]
@jeroenwienk 

Is there any chance this will be implemented anytime soon? I now have to use a dirty work around and create flow cards to allow users to change settings which I'm not able to add to the device itself because I cant add these device settings programmatically.

**Comment by WeeJeWel** [comment]
Don't count on it for now. There are many advantages over statically defined settings, for example, we'd like to show these in the App Store.

Maybe if you share your problem/usecase we can think of a better solution.

**Comment by jghaanstra** [comment]
Bummer, but let me explain ...

The Shelly app supports around 50 different device models on three different protocols (WiFi, Cloud and recently the first Bluetooth devices). Some time back (around the time I created this issue) I switched to generic drivers per protocol to support all models based on their protocol as it would make it a lot easier to manage. This all works well and I can add new devices to the app by mostly only editing the JSON array which holds the configuration of each device. When pairing, the device specifics are added using  `Homey.createDevice()` and pulled from this JSON array.

The only thing missing is that I'm unable to programmatically define is the access to the device settings. I can add the settings themself but I can not define how they can be accessed under the device. For WiFi and cloud devices that has never been much of an issue as I could use a standard template for all devices. There are only a couple of devices that required some device specific user settings which I implemented as action cards.

But now when adding a generic driver for Bluetooth with the first two Bluetooth devices this is becoming an issue again as both these devices require device specific user settings (a beacon out of range for a Bluetooth button and an invert alarm_contact for a Bluetooth DW sensor).

As out-of-the-box alternative the only other implementation I could come up with is having a filter option in `driver.settings.compose.json` based on capabilities. But interested to hear if Athom is seeing other options for this.

**Comment by WeeJeWel** [comment]
I understand, thank you for sharing.

I gotta be honest though, the real solution would be to create a driver per device type, which imho is a better solution for the end-user as well, because they show up in the Homey App Store as separate identities as well.

For now, we'll keep your request in mind and think about it in future plans!

**Comment by frodeheg** [comment]
@WeeJeWel , one problem with not being able to change the settings programmatically is as follows:

* Some products can have many firmware revisions. The older firmware revisions do not expose all the features of the device. In my case these features are not capabilities, so I expose them as settings, and since settings cannot be added/removed like addCapability/removeCapability then it appears to the user that some settings are available but in fact, they are not.
